include/magma/vk/textures/Texture.h

```diff
@@ -0,0 +1,40 @@
+/**
+ * @file Texture.h
+ * @brief Class contain all information about texture
+ * @version 0.1
+ * @date 2021-02-12
+ */
+#pragma once
+#include <string>
+#include <vulkan/vulkan.h>
+#include <memory>
+
+#include "magma/vk/textures/ImageView.h"
+
+struct TextureInfo
+{
+    VkImageCreateInfo       imageInfo;
+    VkImageViewCreateInfo   viewInfo;
+    VkImageLayout           curLayout;
+    std::string             name;
+};
+
+class Texture {
+public:
+    Texture();
+    Texture(VkImage img, VkDeviceMemory mem, ImageView defaultImageView, TextureInfo* info);
+
+    const VkImage           &getImage()     const { return _image;                      }
+    const VkDeviceMemory    &getMemory()    const { return _imageMemory;                }
+    const VkImageView       &getView()      const { return _defaultImageView.getView(); }
+    const ImageView         &getImageView() const { return _defaultImageView;           }
+    TextureInfo*             getInfo()      const { return _info;                       }
+
+private:
+    VkDevice _device;
```

Comment:
I guess _device should be a part of TextureInfo

Reply:
Moreover, _device member is always uninitialized now

Reply:
Done